apps/docs/src/app/examples/switches/CustomSwitch.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+import {
+  HvBaseSwitch,
+  HvButtonBase,
+  HvTypography,
+} from "@hitachivantara/uikit-react-core";
+
+export default function Demo() {
+  const [checked, setChecked] = useState(false);
+
+  return (
+    <HvButtonBase
+      className="flex items-center gap-xs border-rounded-full p-l-xxs w-74px justify-between h-26px [&:hover_.HvBaseSwitch-root]:bg-transparent [&_.HvBaseSwitch-root:hover]:bg-transparent"
+      onClick={() => setChecked((prev) => !prev)}
```

Comment:
we can't have interactive elements inside other interactive elements. this needs to be something like this:
```suggestion
      component="div"
      onClick={() => switchRef.current.click()}
```